nl_server/loader.py

```diff
@@ -45,6 +47,9 @@ def load_server_state(app: Flask):
   flask_env = os.environ.get('FLASK_ENV')
 
   embeddings_dict = _load_yaml(flask_env)
+  vertex_ai_models = {}
```

Comment:
I find it a bit hard to following load_server_state process here. As I mentioned in the other comment, maybe we should have one yaml file to hold all the model/embedding information. So this function would read the yaml and switch between the "type" and process?

Imaging if we decide to use huggingface endpoint or other 3p model endpoint. The change would just be add a new switch path in here. But now there are "vertext_ai" specific naming, functions and logic route. It would not be easy to add a new type of endpoint.

Reply:
chatted & will chat more offline about changes to make after this PR goes in